File: LeetCode-Solutions/1694-make-sum-divisible-by-p/solution.cpp
class Solution {
public:
    int minSubarray(vector<int>& nums, int p) {
        int n = nums.size();
        long long sum_n = 0;
        for ( auto i : nums ) {
            sum_n += i;
        }
        sum_n %= p;
        if ( sum_n == 0 ) return 0;
        long long sum = 0;
        int ans = 1e9;
        map<long long,int> mp;
        mp[sum_n] = 1;
        for ( int i = 0 ; i < n ; ++i ) {
            sum += nums[i];
            sum %= p;
            if ( mp[sum] != 0 + (i == n - 1) ) {
                ans = min(ans,i-mp[sum]+2);
            }
            mp[(sum + sum_n)%p] = i + 2;
        }
        if ( ans == 1e9 ) ans = -1;
        return ans;
    }
};

File: LeetCode-Solutions/1694-make-sum-divisible-by-p/solution.py
class Solution:
    def minSubarray(self, nums: List[int], p: int) -> int:
        n = len(nums)
        sum = [0]*n
        sum[0] = nums[0]
        for i in range(1,n):
            sum[i] = sum[i-1] + nums[i]
        
        mark = {}
        mark[0] = n - 1
        ans = n
        for i in range(n - 1,-1,-1):
            if ( (sum[n - 1] - sum[i] + nums[i]) %p == 0 ):
                ans = min(ans,i)
            y = (p - sum[i]%p)%p
            if ( y in mark ):
                ans = min(ans,mark[y] - i)
            x = (sum[n-1] - sum[i])%p
            mark[x] = i

        if (ans == n):
            ans = -1
        return ans

